packages/core/src/telemetry/clearcut-logger/clearcut-logger.test.ts

```diff
@@ -4,203 +4,249 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { vi, describe, it, expect, beforeEach, afterEach } from 'vitest';
-import * as https from 'https';
-import { ClientRequest, IncomingMessage } from 'http';
-import { Readable, Writable } from 'stream';
+import {
+  vi,
+  describe,
+  it,
+  expect,
+  afterEach,
+  beforeAll,
+  afterAll,
+} from 'vitest';
 
 import {
   ClearcutLogger,
-  LogResponse,
   LogEventEntry,
+  TEST_ONLY,
 } from './clearcut-logger.js';
-import { Config } from '../../config/config.js';
+import { ConfigParameters } from '../../config/config.js';
 import * as userAccount from '../../utils/user_account.js';
 import * as userId from '../../utils/user_id.js';
+import { EventMetadataKey } from './event-metadata-key.js';
+import { makeFakeConfig } from '../../test-utils/config.js';
+import { http, HttpResponse } from 'msw';
+import { server } from '../../mocks/msw.js';
 
-// Mock dependencies
-vi.mock('https-proxy-agent');
-vi.mock('https');
 vi.mock('../../utils/user_account');
 vi.mock('../../utils/user_id');
 
-const mockHttps = vi.mocked(https);
 const mockUserAccount = vi.mocked(userAccount);
 const mockUserId = vi.mocked(userId);
 
+// TODO(richieforeman): Consider moving this to test setup globally.
+beforeAll(() => {
+  server.listen({});
+});
+
+afterEach(() => {
+  server.resetHandlers();
+});
+
+afterAll(() => {
+  server.close();
+});
+
 describe('ClearcutLogger', () => {
-  let mockConfig: Config;
-  let logger: ClearcutLogger | undefined;
+  const NEXT_WAIT_MS = 9001;
```

Comment:
why 9001. Surprising choice for a magic number. Comment or make something less seemingly precise.